packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -145,57 +141,111 @@ fn storage_type(attrs: &[Attribute]) -> StorageType {
     .expect("Storage type must be specified exactly once as 'instance', 'persistent', or 'temporary'.")
 }
 
-/// Returns the value type of a storage enum variant.
-fn value_type(attrs: &[Attribute]) -> Type {
-    attrs
+/// Returns the status xor value type of a storage enum variant.
+fn value(attrs: &[Attribute]) -> Value {
+    let has_status = attrs.iter().any(|attr| attr.path().is_ident("status"));
+    let value_attrs: Vec<_> = attrs
```

Comment:
I would keep the previous approach with flat_map + exactly_one. If you don't like the `match{ _ if ... =>...}` pattern you can also use `if {} else if {} else{}`

for example:
```suggestion
fn value(attrs: &[Attribute]) -> Value {
    attrs.iter().flat_map(|attr| match attr.meta{
        _ if attr.path().is_ident("status")) => Some(Value:Status),
        Meta::List(list) if attr.path().is_ident("value")) => Some(Value::Type(list.parse_args::<Type>().expect("Failed to parse value type."))),
        _ => None,
        }).
        .exactly_one()
        .expect("each storage key must either have a #[status] xor #[value(Type)] attribute")
```

This is a little less precise on the error when you have a value attribute, but no `Type`, so you could also split that into two cases if you want to preserve that panic